addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -6975,10 +6971,57 @@ registry.ImageTools = ImageHandlerOption.extend({
      */
     async _writeShape(svgText) {
         const img = this._getImg();
-        const initialImageWidth = img.naturalWidth;
+        const svg = new DOMParser().parseFromString(svgText, 'image/svg+xml').documentElement;
         let needToRefreshPublicWidgets = false;
         let hasHoverEffect = false;
 
+        // Add shape animations on hover.
+        if (img.dataset.hoverEffect && this._canHaveHoverEffect()) {
+            await this._addImageShapeHoverEffect(svg, img);
+            // The "ImageShapeHoverEffet" public widget needs to restart
+            // (e.g. image replacement).
+            needToRefreshPublicWidgets = true;
+            hasHoverEffect = true;
+        }
+
+        const dataURL = await this.computeShape(svgText, img);
```

Comment:
Maybe replace `svgText` by the `svg` so you can avoid parsing the SVG again in `computeShape`. (to be tested)